### PR TITLE
Specialist Unit View LESQ-500

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: {owa_music_year}

- Adds specialist unit view, fixture and stories file

## Issue(s)

Fixes #LESQ-500

## How to test

1. Go to https://deploy-preview-2187--owa-storybook.netlify.thenational.academy/?path=/story/components-teacherviews-specialistunitlisting--specialist-unit-listing-page

## Screenshots
Screenshot 2024-01-17 at 12 08 49

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/pages/teachers/programmes/[programmeSlug]/units.tsx [hunk 1]
```diff
@@ -98,16 +99,27 @@ const UnitListingPage: NextPage<UnitListingPageProps> = ({
...
+    ...props
+  }: UnitListItemProps | IndividualSpecialistUnit) => {
+    // Temporary until tracking for specialist units
+    const isSpecialistUnit = (
+      x: UnitListItemProps | IndividualSpecialistUnit,
+    ): x is IndividualSpecialistUnit => {
+      return "developmentalStageTitle" in x;
+    };
+
+    if (!isSpecialistUnit(props)) {
```
Comment (njblore) [inline src/pages/teachers/programmes/[programmeSlug]/units.tsx:112]: specialist pages will be in a different folder/route (`specialistProgrammes/[programmeSlug]/specialistUnits`) so this page shouldn't need to worry about there being any specialist data

File: src/components/UnitAndLessonLists/ListItemHeader.tsx [hunk 2]
```diff
@@ -36,6 +37,22 @@ type ListItemHeadingProps = CommonProps &
...
   };
 
+type SpecialistListItemProps = IndividualSpecialistUnit &
+  CommonProps & {
+    title: LessonListItemProps["lessonTitle"] | UnitListItemProps["title"];
+    slug: LessonListItemProps["lessonSlug"] | UnitListItemProps["slug"];
+    expired: boolean | null;
+    index?: number;
+    isExemplarUnit?: boolean;
+    yearTitle?: string | null;
```
Comment (njblore) [inline src/components/UnitAndLessonLists/ListItemHeader.tsx:47]: specialist doesn't have years or exemplar status

File: src/components/OptionalityCard/OptionalityCard.tsx [hunk 3]
```diff
@@ -4,17 +4,22 @@ import Box from "@/components/SharedComponents/Box";
...
+import {
+  IndividualSpecialistUnit,
+  SpecialistUnit,
+} from "@/components/TeacherViews/SpecialistUnitListing/SpecialistUnitListing.view";
 
 export type UnitOption = Omit<UnitData, "unitStudyOrder">;
 
 type OptionalityCardProps = {
-  unitOptions: UnitOption[];
+  unitOptions: UnitOption[] | SpecialistUnit;
```
Comment (njblore) [inline src/components/OptionalityCard/OptionalityCard.tsx:20]: specialist won't have optionality right?

File: src/components/TeacherViews/SpecialistUnitListing/SpecialistUnitListing.view.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+  programmeSlug: string;
+  subjectSlug: string;
+  subjectTitle: string;
+  lessonCount: number;
+  unitStudyOrder: number;
+  expired: boolean;
+  expiredLessonCount: number;
+  themeSlug: string | null;
+  themeTitle: string | null;
+  developmentalStageSlug: string;
```
Comment (njblore) [inline src/components/TeacherViews/SpecialistUnitListing/SpecialistUnitListing.view.tsx:50]: I think developmentalStage needs to be optional (therapies have none, possibly not all specialist does)

File: src/components/TeacherComponents/UnitList/UnitList.tsx [hunk 5]
```diff
@@ -21,23 +27,37 @@ export type Tier = {
...
         <>
           <UL aria-label="A list of units" $reset>
             {currentPageItems.map((item, index) => (
-              <LI key={`UnitList-UnitListItem-${item[0]?.slug}`}>
-                {item.length > 1 ? (
+              <LI
+                key={`UnitList-UnitListItem-${item[0]?.slug}`}
+                data-testid="unit-list-item"
+              >
+                {item.length > 1 && isUnitOption(item) ? (
```
Comment (JBR90) [inline src/components/TeacherComponents/UnitList/UnitList.tsx:60]: Do some specialist items have more than one pathway at this point?

**Review by JBR90 — APPROVED** [review]
Approved with question

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
- [x] The subject name is missing in the breadcrumb, at this point is should be displayed (and not clickable)
- [ ] When at tablet widths, the programme options stack on top of each other. In Nirav's design they go side by side, and I assume, scrollable (at the minute if you squash down Maths enough, the options bleed off the edge of the page badly https://www.thenational.academy/teachers/programmes/maths-secondary-ks4-foundation-l/units)

**Review by jack-skerrett-oak — APPROVED** [review]
We have made this ticket to cover the work needed to add in a scrollable component at mobile widths https://www.notion.so/oaknationalacademy/I-want-a-horizontal-scroll-for-when-programme-names-become-to-long-on-the-subject-listing-page-4fa923916f2e4261899b718760310fc9?pvs=4

**Pull request merged**